File: backend/apps/media_crawl/schemas/report.py
# -*- coding: utf-8 -*-
"""Schemas for Analysis Report API."""

from datetime import datetime
from typing import List, Optional
from uuid import UUID

from ninja import Schema


class SourceKeywordsResponse(Schema):
    """Response schema for source keywords list."""

    keywords: List[str]


class ExportDataRequest(Schema):
    """Request schema for data export."""

    platform: str
    source_keyword: str
    time_from: Optional[int] = None
    time_to: Optional[int] = None


class ExportDataResponse(Schema):
    """Response schema for data export."""

    csv_data: str
    record_count: int
    truncated: bool = False
    total_count: int = 0


class ReportCreateRequest(Schema):
    """Request schema for creating a report."""

    title: str
    platform: str
    source_keyword: str
    time_from: Optional[int] = None
    time_to: Optional[int] = None
    content: str
    record_count: int = 0


class ReportSchema(Schema):
    """Response schema for a single report."""

    id: UUID
    title: str
    platform: str
    source_keyword: str
    time_from: Optional[int] = None
    time_to: Optional[int] = None
    content: str
    record_count: int
    created_at: datetime


class ReportListItemSchema(Schema):
    """Response schema for report list item (without content)."""

    id: UUID
    title: str
    platform: str
    source_keyword: str
    time_from: Optional[int] = None
    time_to: Optional[int] = None
    record_count: int
    created_at: datetime


class ReportListResponse(Schema):
    """Paginated response for report list."""

    items: List[ReportListItemSchema]
    total: int
    limit: int
    offset: int


class ReportErrorResponse(Schema):
    """Error response for report endpoints."""

    detail: str
    code: str

File: frontend/src/types/report.ts
/**
 * Types for Analysis Report feature.
 */

export interface SourceKeywordsResponse {
  keywords: string[]
}

export interface ExportDataRequest {
  platform: string
  source_keyword: string
  time_from?: number | null
  time_to?: number | null
}

export interface ExportDataResponse {
  csv_data: string
  record_count: number
  truncated: boolean
  total_count: number
}

export interface ReportCreateRequest {
  title: string
  platform: string
  source_keyword: string
  time_from?: number | null
  time_to?: number | null
  content: string
  record_count: number
}

export interface Report {
  id: string
  title: string
  platform: string
  source_keyword: string
  time_from?: number | null
  time_to?: number | null
  content: string
  record_count: number
  created_at: string
}

export interface ReportListItem {
  id: string
  title: string
  platform: string
  source_keyword: string
  time_from?: number | null
  time_to?: number | null
  record_count: number
  created_at: string
}

export interface ReportListResponse {
  items: ReportListItem[]
  total: number
  limit: number
  offset: number
}

export interface ReportError {
  detail: string
  code: string
}

